packages/ui/SideNavigationBar/index.tsx

```diff
@@ -72,6 +75,7 @@ export const SideNavigationBar = ({
     setSubMenuId(id);
     resetIsNavListItemClicked();
     setIsSubMenuShown(true);
+    resetSearch();
   };
```

Comment:
Here, when you reset it after opening the submenu, you force an additional request and refresh the list, but the search state is the same (empty string). Setting `resetSearch();` in `hideSubMenu()` I think is enough. Or is there some other use case that I don't see?
```suggestion
    setSubMenuId(id);
    resetIsNavListItemClicked();
    setIsSubMenuShown(true);
  };
```

Reply:
I use it to refresh budget list when you are in specific budget after search. Number 2 in my PR describe.

Reply:
Okay now I see, my mistake :)
I'm still wondering, because when you reopen after changing the budget for a second, you will see the old searched budget 🤔

Reply:
Probably debounce. I’ll try to do something with it

Reply:
Previously, when the cleanup worked, there was a useEffect in the SubMenu/index.tsx file that called onChange on the input with an empty string when the component was unmounted. I'm not sure if this was the best solution:
```tsx
  useEffect(() => {
    return () => {
      searchInput?.onChange?.("");
    };
  }, []);
```

Reply:
Yes but with it there was ESlint warning and probably (I don't really remember) it don't work in case described in number 2 in PR description.

Reply:
It works, but you are right about the ESlint warning caused by the empty dependency array in useEffect.